lib/query.js

```diff
@@ -232,7 +232,7 @@ export const getCoefficientsAndConstants = (node, coefficientMap = {}, constants
         } else {
             coefficientMap[key].push(coefficient)
         }
-    } else if(isPolynomial(node) || isApply(node)) {
+    } else if(isPolynomial(node) || isAdd(node) || (isMul(node) && node.args.every(isPolynomialTerm))) {
         // 2x^2 + 3x + 1, 3x^2 * 2x^2
```

Comment:
Given the comment, shouldn't checking `isPolynomial(node)` enough since we've already ruled out constants and single term polynomials.  Why are the other conditions necessary?